lib/sycamore/sycamore/transforms/extract_document_structure.py

```diff
@@ -211,3 +255,13 @@ def summarize_sections(doc: HierarchicalDocument) -> HierarchicalDocument:
                 summary_list.append(elem_sum)
             section.data["summary"] = "".join(summary_list)
         return doc
+
+
+class ExtractImageSummaries(ExtractSummary):
+    @staticmethod
+    def summarize(doc: HierarchicalDocument) -> HierarchicalDocument:
+        for section in doc.children:
+            assert len(section.children) == 1 and section.children[0].type == "Image"
```

Comment:
Maybe this can be more robust by doing something like 
```python
for child in section.children:
    if child.type == "Image":
        continue
```
?

Reply:
Sooo the thing is this function actually expects only one image per section when i was coding it up last week. This is screaming technical debt and I don't want to leave this code here. I have a meeting right now, but I'll stop by after to see if we can talk about what we can do to make these functions not finetuned to one specific use case.